### PR TITLE
SRS Symptoms flow (EXPOSUREAPP-14179)

### PR DESCRIPTION
**Pull request by SamuraiKek**
- To test, also add some check-ins. 
- Because there is no thank you screen yet, you will land on the home screen once you complete the submission.
- Also check the journal once you have completed the submission
- Make sure that the cancel submission dialog is shown when you go back or press X on the check-ins screen, symptom intro screen and symptom calendar screen.
- Make sure that selecting no/no answer on the symptoms intro screen will show the submission warning dialog

- [ ] Thank you screen was merged -> add navigation to it.
- [ ] Error codes were merged -> add error dialog.

### CODE REVIEW

**Comment by mtwalli** [comment]
@InkenTownsend  Please have a look?

**Review by InkenTownsend — APPROVED** [review]
UA Approved

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/symptoms/SrsSymptomsCalendarViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        selectedCheckIns?.let {
+            checkInRepository.updateSubmissionConsents(
+                checkInIds = it.asList(),
+                consent = true
+            )
+        }
+        srsSubmissionRepository.submit(
+            testType,
+            Symptoms(startOfSymptoms = symptomStartInternal.value, symptomIndication = symptomsIndication)
+        )
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/symptoms/SrsSymptomsCalendarViewModel.kt:64]: you need to handle errors here, can throw an error

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/symptoms/SrsSymptomsIntroductionViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        selectedCheckIns?.let {
+            checkInRepository.updateSubmissionConsents(
+                checkInIds = it.asList(),
+                consent = true
+            )
+        }
+
+        srsSubmissionRepository.submit(
+            testType,
+            Symptoms(startOfSymptoms = null, symptomIndication = symptomsIndication)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/symptoms/SrsSymptomsIntroductionViewModel.kt:81]: same here

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/checkins/SrsCheckinsFragmentViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    fun onSkipClick() {
+        Timber.d("onSkipClick")
+        events.postValue(SrsCheckinsNavigation.ShowSkipDialog)
+    }
+
+    fun onNextClick() {
+        Timber.d("onNextClick")
+        val idsWithConsent = selectedSetFlow.value
+        events.postValue(SrsCheckinsNavigation.GoToSymptomSubmission(testType, idsWithConsent.toLongArray()))
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/checkins/SrsCheckinsFragmentViewModel.kt:64]: you don't need to pass the ids around, it should be persisted in the repo and it is accessed from repo

**Review by mtwalli — APPROVED** [review]

**Pull request merged**